src/compiler/collections.ts

```diff
@@ -32,20 +33,18 @@ namespace ts {
         However, `forEach` will iterate over strings because values are stringified before being put in the map.
         */
 
-        constructor(pairs?: [K, V][]) {
-            if (pairs) {
-                for (const [key, value] of pairs) {
-                    this.data[key as string] = value;
-                }
-            }
-        }
+        constructor() {}
 
         clear(): void {
             this.data = createDictionaryModeObject<V>();
         }
 
-        delete(key: K): void {
-            delete this.data[key as string];
+        delete(key: K): boolean {
+            const had = this.has(key);

```

Comment:
`delete` [always](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete) returns `true` in strict mode.